### PR TITLE
APP-1583 / Remove setup ABIs

### PR DESCRIPTION
**Pull request by Rekard0**
## Description

Task: [APP-1583](APP-1583)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

[APP-1583]: https://aragonassociation.atlassian.net/browse/APP-1583?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/contract-tests.yml [hunk 1]
```diff
@@ -32,3 +32,9 @@ jobs:
         env:
           REPORT_GAS: true
         run: yarn run test
+      - name: Test deployment (hardhat)
+        run: yarn deploy --network hardhat --reset
+        env:
+          ETH_KEY: ''
+          HARDHAT_DAO_ENS_DOMAIN: 'dao.eth'
+          HARDHAT_PLUGIN_ENS_DOMAIN: 'plugin.eth'
```
Comment (brickpop) [inline .github/workflows/contract-tests.yml:40]: This should be either `dao-plugin.eth`  or  `plugin.dao.eth`

File: packages/contracts/contracts/plugin/examples/count/CounterV1.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (brickpop) [inline packages/contracts/contracts/plugin/examples/count/CounterV1.json:1]: - Shouldn't this be called `metadata.json` as well?
- I would add `name: "...", description: ""` everywhere
- `setupABIs` => `pluginSetupABI:  {prepareInstallation, prepareUpdate, prepareUninstallation}`

Comment (novaknole) [inline packages/contracts/contracts/plugin/examples/count/CounterV1.json:1]: This wont' work well. It should have a different structure, where each field could hold description as well.  Probably a separate task @brickpop but we need to design this before publishing plugins to our plugin repos.

File: packages/contracts/contracts/plugin/examples/count/CounterV2.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "setupABIs": {
```
Comment (novaknole) [inline packages/contracts/contracts/plugin/examples/count/CounterV2.json:2]: same here.

File: packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol [hunk 4]
```diff
@@ -16,12 +16,15 @@ contract PluginCloneableSetupV1Mock is PluginSetup {
     }
 
     /// @inheritdoc IPluginSetup
-    function prepareInstallationDataABI() external view virtual override returns (string memory) {
-        return "";
-    }
+    // function prepareInstallationDataABI() external view virtual override returns (string memory) {
```
Comment (novaknole) [inline packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol:19]: needs to be removed.

File: packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol [hunk 5]
```diff
@@ -16,12 +16,15 @@ contract PluginCloneableSetupV1Mock is PluginSetup {
...
+    //     return "";
+    // }
```
Comment (brickpop) [inline packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol:21]: Why not removed?

File: packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol [hunk 6]
```diff
@@ -37,9 +40,9 @@ contract PluginCloneableSetupV1Mock is PluginSetup {
     }
 
     /// @inheritdoc IPluginSetup
-    function prepareUninstallationDataABI() external view virtual override returns (string memory) {
-        return "";
-    }
+    // function prepareUninstallationDataABI() external view virtual override returns (string memory) {
+    //     return "";
+    // }
```
Comment (brickpop) [inline packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol:46]: Same

File: packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol [hunk 7]
```diff
@@ -37,9 +40,9 @@ contract PluginCloneableSetupV1Mock is PluginSetup {
     }
 
     /// @inheritdoc IPluginSetup
-    function prepareUninstallationDataABI() external view virtual override returns (string memory) {
-        return "";
-    }
+    // function prepareUninstallationDataABI() external view virtual override returns (string memory) {
```
Comment (novaknole) [inline packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol:43]: remove.

File: packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol [hunk 8]
```diff
@@ -64,12 +67,15 @@ contract PluginCloneableSetupV2Mock is PluginCloneableSetupV1Mock {
     }
 
     /// @inheritdoc IPluginSetup
-    function prepareInstallationDataABI() external view virtual override returns (string memory) {
-        return "";
-    }
+    // function prepareInstallationDataABI() external view virtual override returns (string memory) {
+    //     return "";
+    // }
```
Comment (brickpop) [inline packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol:72]: Same

File: packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol [hunk 9]
```diff
@@ -64,12 +67,15 @@ contract PluginCloneableSetupV2Mock is PluginCloneableSetupV1Mock {
     }
 
     /// @inheritdoc IPluginSetup
-    function prepareInstallationDataABI() external view virtual override returns (string memory) {
-        return "";
-    }
+    // function prepareInstallationDataABI() external view virtual override returns (string memory) {
```
Comment (novaknole) [inline packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol:70]: remove.

File: packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol [hunk 10]
```diff
@@ -85,9 +91,9 @@ contract PluginCloneableSetupV2Mock is PluginCloneableSetupV1Mock {
     }
 
     /// @inheritdoc IPluginSetup
-    function prepareUninstallationDataABI() external view virtual override returns (string memory) {
-        return "";
-    }
+    // function prepareUninstallationDataABI() external view virtual override returns (string memory) {
```
Comment (novaknole) [inline packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol:94]: remove.

File: packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol [hunk 11]
```diff
@@ -85,9 +91,9 @@ contract PluginCloneableSetupV2Mock is PluginCloneableSetupV1Mock {
...
+    //     return "";
+    // }
```
Comment (brickpop) [inline packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol:96]: Same

File: packages/contracts/contracts/test/plugin-mock/UUPSUpgradeable/PluginUUPSUpgradeableSetupMock.sol [hunk 12]
```diff
@@ -16,12 +16,15 @@ contract PluginUUPSUpgradeableSetupV1Mock is PluginSetup {
     }
 
     /// @inheritdoc IPluginSetup
-    function prepareInstallationDataABI() external view virtual override returns (string memory) {
-        return "";
-    }
+    // function prepareInstallationDataABI() external view virtual override returns (string memory) {
+    //     return "";
+    // }
```
Comment (brickpop) [inline packages/contracts/contracts/test/plugin-mock/UUPSUpgradeable/PluginUUPSUpgradeableSetupMock.sol:21]: Same

File: packages/contracts/contracts/test/plugin-mock/UUPSUpgradeable/PluginUUPSUpgradeableSetupMock.sol [hunk 13]
```diff
@@ -37,9 +40,9 @@ contract PluginUUPSUpgradeableSetupV1Mock is PluginSetup {
     }
 
     /// @inheritdoc IPluginSetup
-    function prepareUninstallationDataABI() external view virtual override returns (string memory) {
-        return "";
-    }
+    // function prepareUninstallationDataABI() external view virtual override returns (string memory) {
+    //     return "";
+    // }
```
Comment (brickpop) [inline packages/contracts/contracts/test/plugin-mock/UUPSUpgradeable/PluginUUPSUpgradeableSetupMock.sol:45]: Same

File: packages/contracts/contracts/test/plugin-mock/UUPSUpgradeable/PluginUUPSUpgradeableSetupMock.sol [hunk 14]
```diff
@@ -16,12 +16,15 @@ contract PluginUUPSUpgradeableSetupV1Mock is PluginSetup {
     }
 
     /// @inheritdoc IPluginSetup
-    function prepareInstallationDataABI() external view virtual override returns (string memory) {
-        return "";
-    }
+    // function prepareInstallationDataABI() external view virtual override returns (string memory) {
```
Comment (novaknole) [inline packages/contracts/contracts/test/plugin-mock/UUPSUpgradeable/PluginUUPSUpgradeableSetupMock.sol:19]: remove commented things from this file completely.

File: packages/contracts/contracts/voting/addresslist/metadata.json [hunk 15]
```diff
@@ -0,0 +1,7 @@
+{
+  "setupABIs": {
```
Comment (novaknole) [inline packages/contracts/contracts/voting/addresslist/metadata.json:2]: same. we need to design a structure for this.

File: packages/contracts/contracts/voting/admin/metadata.json [hunk 16]
```diff
@@ -0,0 +1,7 @@
+{
+  "setupABIs": {
+    "prepareInstallationDataABI": "(address admin)",
```
Comment (novaknole) [inline packages/contracts/contracts/voting/admin/metadata.json:3]: same. I won't comment anymore about this. as every JSON needs better structure.

File: packages/contracts/contracts/voting/multisig/MultisigSetup.sol [hunk 17]
```diff
@@ -40,20 +38,15 @@ contract MultisigSetup is PluginSetup {
         IDAO dao = IDAO(_dao);
 
         // Decode `_data` to extract the params needed for deploying and initializing `Multisig` plugin.
-        (
-            address[] memory members,
-            Multisig.MultisigSettings memory multisigSettings
-        ) = abi.decode(_data, (address[], Multisig.MultisigSettings));
+        (address[] memory members, Multisig.MultisigSettings memory multisigSettings) = abi.decode(
```
Comment (novaknole) [inline packages/contracts/contracts/voting/multisig/MultisigSetup.sol:41]: are you using formatter prettier ?

File: packages/contracts/deploy/helpers.ts [hunk 18]
```diff
@@ -2,6 +2,7 @@ import {promises as fs} from 'fs';
 import {ethers} from 'hardhat';
 import {BigNumberish} from 'ethers';
 import {HardhatRuntimeEnvironment} from 'hardhat/types';
+import IPFS from 'ipfs-http-client';
```
Comment (brickpop) [inline packages/contracts/deploy/helpers.ts:5]: FYI: We could also use our own IPFS cluster client `@aragon/sdk-ipfs`

File: packages/contracts/deploy/00_managing-dao/00_managing-dao.ts [hunk 19]
```diff
@@ -23,7 +23,7 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
       execute: {
         init: {
           methodName: 'initialize',
-          args: ['0x00', deployer, ethers.constants.AddressZero],
+          args: ['0x00', deployer, ethers.constants.AddressZero, '0x'],
```
Comment (novaknole) [inline packages/contracts/deploy/00_managing-dao/00_managing-dao.ts:26]: add comment here that says what this is, mark it with TODO: so we know we might need to update it.

File: packages/contracts/CHANGELOG.md [hunk 20]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Added `metadata.json` for all plugins.
```
Comment (brickpop) [inline packages/contracts/CHANGELOG.md:12]:
```suggestion
- Added a `metadata.json` file for all plugins.
```

File: packages/contracts/deploy/30_plugins/10_plugin-repos/00_create-plugins-repo.ts [hunk 21]
```diff
@@ -10,29 +15,61 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
+    network.name
+  );
   await createPluginRepo(
     hre,
     'AddresslistVoting',
     'AddresslistVotingSetup',
     [1, 0, 0],
-    '0x'
+    ethers.utils.hexlify(
+      ethers.utils.toUtf8Bytes(`ipfs://${addresslistCIDPath}`)
```
Comment (novaknole) [inline packages/contracts/deploy/30_plugins/10_plugin-repos/00_create-plugins-repo.ts:31]: probably, @mathewmeconry can better review this part until i remember ipfs

File: packages/contracts/test/voting/addresslist-voting-setup.ts [hunk 22]
```diff
@@ -94,9 +95,7 @@ describe('AddresslistVotingSetup', function () {
       const dataHRABI =
```
Comment (novaknole) [inline packages/contracts/test/voting/addresslist-voting-setup.ts:95]: why is this here ?

File: packages/contracts/contracts/plugin/examples/count/CounterV2.json [hunk 23]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (brickpop) [inline packages/contracts/contracts/plugin/examples/count/CounterV2.json:1]: metadata.json for clarity

File: packages/contracts/test/voting/addresslist-voting-setup.ts (hunk 22 above)
Comment (novaknole) [reply packages/contracts/test/voting/addresslist-voting-setup.ts:95]: It seems this checks don't make sense anymore.

**Review by novaknole — COMMENTED** [review]
After going through the reviews, one more thing.

In PSP, IPluginSetup, DAOFactory,  there're comments mentioning what `data` is. It's mostly said `.....dataABI` function or `.....dataABI` in the plugin setup contracts. Those are not relevant and should mention `json metadata`.

File: packages/contracts/deploy/30_plugins/10_plugin-repos/00_create-plugins-repo.ts [hunk 24]
```diff
@@ -1,7 +1,12 @@
 import {HardhatRuntimeEnvironment} from 'hardhat/types';
 import {DeployFunction} from 'hardhat-deploy/types';
 
-import {createPluginRepo} from '../../helpers';
+import addresslistMetadata from '../../../contracts/voting/addresslist/metadata.json';
```
Comment (brickpop) [inline packages/contracts/deploy/30_plugins/10_plugin-repos/00_create-plugins-repo.ts:4]: Please, let's double check the task description

We need plugins to have two different "metadata's":
- Per plugin (global)
  - Name
  - Description
  - Screenshots, etc
- Per version
  - Changes
  - ABI's

File: packages/contracts/contracts/plugin/examples/count/CounterV1.json (hunk 2 above)
Comment (novaknole) [reply packages/contracts/contracts/plugin/examples/count/CounterV1.json:1]: `name`, `description` + 

putting this dirrectly mightn't work:

```
(tuple(uint8 votingMode, uint64 supportThreshold, uint64 minParticipation, uint64minDuration, uint256 minProposerVotingPower) votingSettings, address[] members
```. Front-end will have a hard time to decode this.

probably, it should have an actual ABI(how it looks normally, or we could do `human readable ABI` from ethers.js.

Needs really good look to make it very nice.

File: packages/contracts/deploy/30_plugins/10_plugin-repos/00_create-plugins-repo.ts (hunk 24 above)
Comment (novaknole) [reply packages/contracts/deploy/30_plugins/10_plugin-repos/00_create-plugins-repo.ts:4]: Let's do `per plugin's release` and not `per plugin`. Thoughts ? @brickpop as each plugin release might have different pictures/information.

File: .github/workflows/contract-tests.yml (hunk 1 above)
Comment (Rekard0) [reply .github/workflows/contract-tests.yml:40]: this will only run on `hardhat` network.
but, amended.

File: packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol (hunk 5 above)
Comment (Rekard0) [reply packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol:21]: amended

File: packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol (hunk 6 above)
Comment (Rekard0) [reply packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol:46]: amended

File: packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol (hunk 8 above)
Comment (Rekard0) [reply packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol:72]: amended

File: packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol (hunk 11 above)
Comment (Rekard0) [reply packages/contracts/contracts/test/plugin-mock/Clonable/PluginCloneableSetupMock.sol:96]: amended

File: packages/contracts/contracts/test/plugin-mock/UUPSUpgradeable/PluginUUPSUpgradeableSetupMock.sol (hunk 12 above)
Comment (Rekard0) [reply packages/contracts/contracts/test/plugin-mock/UUPSUpgradeable/PluginUUPSUpgradeableSetupMock.sol:21]: amended

File: packages/contracts/contracts/test/plugin-mock/UUPSUpgradeable/PluginUUPSUpgradeableSetupMock.sol (hunk 13 above)
Comment (Rekard0) [reply packages/contracts/contracts/test/plugin-mock/UUPSUpgradeable/PluginUUPSUpgradeableSetupMock.sol:45]: amended

File: packages/contracts/CHANGELOG.md (hunk 20 above)
Comment (Rekard0) [reply packages/contracts/CHANGELOG.md:12]: amended

File: packages/contracts/deploy/00_managing-dao/00_managing-dao.ts (hunk 19 above)
Comment (Rekard0) [reply packages/contracts/deploy/00_managing-dao/00_managing-dao.ts:26]: converted to params so it won't need comments.

File: packages/contracts/contracts/voting/multisig/MultisigSetup.sol (hunk 17 above)
Comment (Rekard0) [reply packages/contracts/contracts/voting/multisig/MultisigSetup.sol:41]: yes

File: packages/contracts/contracts/plugin/examples/count/CounterV1.json (hunk 2 above)
Comment (Rekard0) [reply packages/contracts/contracts/plugin/examples/count/CounterV1.json:1]: @brickpop setupABIs => pluginSetupABI is amended.
@novaknole it is already "full" format human readable abi, or i am missing somthing

File: packages/contracts/contracts/plugin/examples/count/CounterV2.json (hunk 23 above)
Comment (Rekard0) [reply packages/contracts/contracts/plugin/examples/count/CounterV2.json:1]: amended

File: packages/contracts/deploy/30_plugins/10_plugin-repos/00_create-plugins-repo.ts (hunk 21 above)
Comment (Rekard0) [reply packages/contracts/deploy/30_plugins/10_plugin-repos/00_create-plugins-repo.ts:31]: amended

File: packages/contracts/test/voting/addresslist-voting-setup.ts (hunk 22 above)
Comment (Rekard0) [reply packages/contracts/test/voting/addresslist-voting-setup.ts]: amended

File: packages/contracts/contracts/plugin/examples/count/v1/metadata.json [hunk 25]
```diff
@@ -0,0 +1,9 @@
+{
+  "ui": "",
+  "change": "",
+  "pluginSetupABI": {
+    "prepareInstallation": "(address multiplyHelper, uint num)",
```
Comment (mathewmeconry) [inline packages/contracts/contracts/plugin/examples/count/v1/metadata.json:5]: I suggest you either add the full function signature `function prepareInstallation(address multiplyHelper, uint num)` or then as an array `["address multiplyHelper", "uint num"]`
The reason is that ethers and other libs either wanna have an array for ABI-De-/Encoding or then the full function signature. With this setup the developer either needs to parse it (which makes handling tuples hard)  or add all the time `function prepareInstallation`

**Review by mathewmeconry — CHANGES_REQUESTED** [review]

File: packages/contracts/contracts/plugin/examples/count/v1/metadata.json (hunk 25 above)
Comment (Rekard0) [reply packages/contracts/contracts/plugin/examples/count/v1/metadata.json]: yes, thanks for the heads up,
the array ABI seems to be more useful for readability and to be used by consumers of the metadata
amended in `[0bfecf2](https://github.com/aragon/core/pull/208/commits/0bfecf27606444e4c03914913f0c90c19a2dcb69)`

File: packages/contracts/deploy/helpers.ts (hunk 18 above)
Comment (Rekard0) [reply packages/contracts/deploy/helpers.ts:5]: yes, tested that,
but the `@aragon/sdk-ipfs` is not designed to be used with node
and it require installing multiple libraries beside it, to make it work,
i am not sure if it worth for this task.

File: packages/contracts/contracts/plugin/examples/count/CounterV1.json (hunk 2 above)
Comment (Rekard0) [reply packages/contracts/contracts/plugin/examples/count/CounterV1.json]: @novaknole amended in [0bfecf2](https://github.com/aragon/core/pull/208/commits/0bfecf27606444e4c03914913f0c90c19a2dcb69)

File: packages/contracts/contracts/plugin/examples/count/v2/metadata.json [hunk 26]
```diff
@@ -2,8 +2,11 @@
   "ui": "",
   "change": "",
   "pluginSetupABI": {
-    "prepareInstallation": "(address multiplyHelper, uint num, uint newVariable)",
-    "prepareUpdate": "(uint _newVariable)",
-    "prepareUninstallation": ""
+    "prepareInstallation": [
+      "address multiplyHelper",
+      "uint num, uint newVariable"
```
Comment (mathewmeconry) [inline packages/contracts/contracts/plugin/examples/count/v2/metadata.json:7]:
```suggestion
      "uint num", "uint newVariable"
```

**Review by mathewmeconry — CHANGES_REQUESTED** [review]

File: packages/contracts/contracts/plugin/examples/count/v2/metadata.json (hunk 26 above)
Comment (Rekard0) [reply packages/contracts/contracts/plugin/examples/count/v2/metadata.json]: amended

**Review by mathewmeconry — DISMISSED** [review]

File: packages/contracts/deploy/30_plugins/10_plugin-repos/00_create-plugins-repo.ts [hunk 27]
```diff
@@ -10,27 +15,57 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
     'TokenVoting',
     'TokenVotingSetup',
-    '0x'
+    ethers.utils.hexlify(ethers.utils.toUtf8Bytes(`ipfs://${tokenCIDPath}`))
   );
 
   // AdminSetup
-  await createPluginRepo(hre, 'Admin', 'AdminSetup', '0x');
+  const adminCIDPath = await uploadToIPFS(
+    JSON.stringify(tokenMetadata),
```
Comment (novaknole) [inline packages/contracts/deploy/30_plugins/10_plugin-repos/00_create-plugins-repo.ts:48]: should be adminMetadata.

File: packages/contracts/deploy/30_plugins/10_plugin-repos/00_create-plugins-repo.ts [hunk 28]
```diff
@@ -10,27 +15,57 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
+    hre,
+    'Admin',
+    'AdminSetup',
+    ethers.utils.hexlify(ethers.utils.toUtf8Bytes(`ipfs://${adminCIDPath}`))
+  );
 
   // MultisigSetup
-  await createPluginRepo(hre, 'Multisig', 'MultisigSetup', '0x');
+  const multisigCIDPath = await uploadToIPFS(
+    JSON.stringify(tokenMetadata),
```
Comment (novaknole) [inline packages/contracts/deploy/30_plugins/10_plugin-repos/00_create-plugins-repo.ts:60]: should be multsig metadata.

**Review by novaknole — COMMENTED** [review]
1. 2 comments from my side about metadata.
2. @mathewmeconry could you review IPFS part ?

File: packages/contracts/deploy/30_plugins/10_plugin-repos/00_create-plugins-repo.ts (hunk 27 above)
Comment (Rekard0) [reply packages/contracts/deploy/30_plugins/10_plugin-repos/00_create-plugins-repo.ts]: amended

File: packages/contracts/deploy/30_plugins/10_plugin-repos/00_create-plugins-repo.ts (hunk 28 above)
Comment (Rekard0) [reply packages/contracts/deploy/30_plugins/10_plugin-repos/00_create-plugins-repo.ts]: amended

**Review by novaknole — APPROVED** [review]

**Review by mathewmeconry — CHANGES_REQUESTED** [review]
Please remove all `TOD:GIORGI` again

**Comment by Rekard0** [comment]
> Please remove all `TOD:GIORGI` again

these are part of develop branch now, does it belong to this PR?

File: .github/workflows/contract-tests.yml [hunk 29]
```diff
@@ -32,3 +32,9 @@ jobs:
         env:
           REPORT_GAS: true
         run: yarn run test
+      - name: Test deployment (hardhat)
+        run: yarn deploy --network hardhat --reset
+        env:
+          ETH_KEY: ''
+          HARDHAT_DAO_ENS_DOMAIN: 'dao.eth'
+          HARDHAT_PLUGIN_ENS_DOMAIN: 'dao-plugin.eth'
```
Comment (brickpop) [inline .github/workflows/contract-tests.yml:40]: Will be `plugin.dao.eth`

**Review by brickpop — APPROVED** [review]
